crates/holochain/src/core/workflow/sys_validation_workflow/validate_op_tests.rs

```diff
@@ -979,13 +1160,978 @@ async fn validate_store_entry_update_changes_entry_type() {
+
+    use holochain_serialized_bytes::prelude::*;
+    use serde::{Deserialize, Serialize};
+    #[derive(Debug, Serialize, Deserialize, SerializedBytes)]
+    struct TestLargeEntry {
+        data: Vec<u8>,
+    }
+
+    let mut test_case = TestCase::new().await;
+
+    // Previous action
+    let mut action = fixt!(Create);
+    action.author = test_case.agent.clone().into();
+    action.timestamp = Timestamp::now();
+    action.action_seq = 10;
+    action.prev_action = fixt!(ActionHash);
+    let previous_action = test_case.sign_action(Action::Create(action)).await;
+
+    // Op to validate
+    let app_entry = Entry::App(AppEntryBytes(
+        TestLargeEntry {
+            data: vec![0; 5_000_000],
+        }
+        .try_into()
+        .unwrap(),
+    ));
+    let entry_hash = EntryHashed::from_content_sync(app_entry.clone());
+    let mut create_action = fixt!(Create);
+    create_action.author = previous_action.action().author().clone();
+    create_action.action_seq = previous_action.action().action_seq() + 1;
+    create_action.prev_action = previous_action.as_hash().clone();
+    create_action.timestamp = Timestamp::now().into();
+    create_action.entry_type = EntryType::App(AppEntryDef::new(
+        0.into(),
+        0.into(),
+        EntryVisibility::Public,
+    ));
+    create_action.entry_hash = entry_hash.as_hash().clone();
+    let op = DhtOp::StoreEntry(
+        fixt!(Signature),
+        holochain_types::action::NewEntryAction::Create(create_action),
+        app_entry,
+    );
+
+    let outcome = test_case
+        .expect_retrieve_records_from_cascade(vec![previous_action])
+        .with_op(op)
+        .run()
+        .await
+        .unwrap();
+
+    assert_eq!(
+        Outcome::Rejected(
+            "The entry size 5000011 was larger than the MAX_ENTRY_SIZE 4000000".to_string()
+        ),
+        outcome,
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn validate_valid_store_entry_update() {
+    holochain_trace::test_run().unwrap();
+
+    let mut test_case = TestCase::new().await;
+
+    // Action to be updated
+    let mut to_update_action = fixt!(Create);
+    to_update_action.author = test_case.agent.clone().into();
+    to_update_action.timestamp = Timestamp::now();
+    to_update_action.action_seq = 5;
+    to_update_action.prev_action = fixt!(ActionHash);
+    to_update_action.entry_type = EntryType::App(AppEntryDef::new(
+        0.into(),
+        0.into(),
+        EntryVisibility::Public,
+    ));
+    to_update_action.entry_hash = fixt!(EntryHash);
+    let to_update_signed_action = test_case
+        .sign_action(Action::Create(to_update_action))
+        .await;
+
+    // Previous action
+    let mut action = fixt!(Create);
+    action.author = test_case.agent.clone().into();
+    action.timestamp = Timestamp::now();
+    action.action_seq = 10;
+    action.prev_action = fixt!(ActionHash);
+    let previous_action = test_case.sign_action(Action::Create(action)).await;
```

Comment:
Why is there this in-between action?

Reply:
For a StoreEntry with an Update action the workflow will check the `prev_action` and the `original_action_address` actions -> https://github.com/holochain/holochain/blob/develop/crates/holochain/src/core/workflow/sys_validation_workflow.rs#L369

That doesn't mean there aren't tests in this file where I'm wrongly providing the prev action where it won't be checked but in this case there's a reason